svc/FetchService.js

```diff
@@ -232,12 +232,28 @@ export class FetchService extends HoistService {
 
         if (!ret.ok) {
             ret.responseText = await this.safeResponseTextAsync(ret);
-            throw Exception.fetchError(opts, ret);
+            const e = Exception.fetchError(opts, ret);
+            if (!XH.isSSO && isRelativeUrl && e.httpStatus === 401) {
+                await this.maybeReloadForAuthAsync();
+            }
+            throw e;
         }
 
         return ret;
     }
 
+    async maybeReloadForAuthAsync() {
+        // Reload app to renew auth. Be careful not to trigger during
+        // initialization, or to get caught in a tight reload loop
+        const {appState, localStorageService} = XH;
+        if (appState !== AppState.RUNNING && appState !== AppState.SUSPENDED) return;
+        if (!olderThan(localStorageService.get('xhLastAuthReload', null), ONE_MINUTE)) return;
+
+        localStorageService.set('xhLastAuthReload', Date.now());
+        XH.reloadApp();
```

Comment:
Maybe the default should be to XH.reloadApp() if not alternate requireAuthHandler method provided?

Reply:
valid point about the big form.  but I hope in practice that's not going to be any more common than, say, losing your work because the app went to sleep.  If the app is not asleep, the session should be kept alive by version checking and other timed processes.   

This code is really designed to catch the case on mobile, where the app has not gone to sleep, but the session with the server has somehow died.  So presumably the case where the user has left their phone app unused for a while. Not sure we need to have high expectations about preserving any data at that point.